design/one-pager-k8s-native-providers.md

```diff
@@ -0,0 +1,692 @@
+kind: KubernetesCluster
+metadata:
+  name: demo-cluster
+  namespace: my-infra-dev
+spec:
+  writeConnectionSecretToRef:
+    name: demo-cluster-creds
+```    
+2. Create a Kubernetes `Provider` whose `credentialsSecretRef` references the
+   `KubernetesCluster` `writeConnectionSecretToRef`.
+```yaml
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: Provider
+metadata:
+  name: demo-k8s-provider
+  namespace: my-infra-dev
+spec:
+  credentialsSecretRef:
+    name: demo-cluster-creds
+```
+3. Provision your Kubernetes resource using the aforementioned `Provider`.
+
+Managing Kubernetes resources in an **existing external cluster**:
+1. Create a Kubernetes `Provider` whose `credentialsSecretRef` references a hand
+   curated `Secret` using the same format as that of a `KubernetesCluster`
+   claim.
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: demo-cluster-creds
+  namespace: my-infra-dev
+type: Opaque
+data:
+  endpoint: MY_K8S_ENDPOINT
+  username: MY_K8S_USERNAME
+  password: MY_K8S_PASSWORD
+  clusterCA: MY_K8S_CLUSTER_CA
+  clientCert: MY_K8S_CLIENT_CERT
+  clientKey: MY_K8S_CLIENT_KEY
+  token: MY_K8S_TOKEN
+---
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: Provider
+metadata:
+  name: demo-k8s-provider
+  namespace: my-infra-dev
+spec:
+  credentialsSecretRef:
+    name: demo-cluster-creds
+```
+2. Provision your Kubernetes resource using the aforementioned `Provider`.
+
+Managing Kubernetes resources in the **Crossplane control cluster**:
+1. Create a Kubernetes `ServiceAccount` and grant it the RBAC permissions
+   required to execute necessary Kubernetes actions.
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: my-service-account
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: my-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  - deployments
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+- apiGroups:
+  - minio.rook.io
+  - cockroachdb.rook.io
+  resources:
+  - "*"
+  verbs:
+  - "*"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: my-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: my-role
+subjects:
+- kind: ServiceAccount
+  name: my-service-account
+  namespace: my-infra-dev
+```
+2. Create a Kubernetes `Provider` whose `credentialsSecretRef` references the
+   aforementioned `ServiceAccount` token.
+```yaml
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: Provider
+metadata:
+  name: demo-k8s-provider
+  namespace: my-infra-dev
+spec:
+  credentialsSecretRef:
+    # Get secret name from service account
+    name: my-service-account-secret
+```
+3. Provision your Kubernetes resource using the aforementioned `Provider`.
+
+## Proposed Workflow for `KubernetesApplication` Resources
```

Comment:
Detailed discussion of changes to `KubernetesApplication` seems a little out of scope given the goals of this design. I wonder whether we should capture this separately - perhaps cut and paste the `KubernetesApplication` bits into a Github issue for us to follow up at some point after this design has been ratified and implemented?

Reply:
Done! #859